dev/release.sh

```diff
@@ -0,0 +1,614 @@
+
+    $VERBOSE "== Comitting updates"
+    git add -u
+    git commit $git_quiet -m "Prepare for $release"
+fi
+
+# Done ###############################################################
+    
+$VERBOSE "== Done"
+
+cat <<EOF
+
+======================================================================
+The release is done, and involves a few commits for you to deal with.
+It has all been done in a clone of this workspace, see details below.
+EOF
+if $do_branch; then
+    cat <<EOF
+Additionally, a release branch has been created for you, so you need
+to look for new commits in two places.
+EOF
+fi
+
+cat <<EOF
+Release worktree:   $release_clone
+master branch:      $current_branch
+EOF
+if $do_branch; then
+    cat <<EOF
+new release branch: $new_branch
+EOF
+fi
+
+cat <<EOF
+======================================================================
+EOF
+
+cat <<EOF
+If something went wrong and you want to start over, all you need is to
+remove the release worktree:
+
+    rm -rf $release_clone
+
+If a tarball was uploaded, you must also clean that away, or ask you
+kind OpenSSL sysadmin to do so.
+EOF
+
+exit 0
+
+### BEGIN MANUAL
+=pod
+
+=head1 NAME
+
+release.sh - OpenSSL release script
+
+=head1 SYNOPSIS
+
+B<release.sh>
+[
+B<--major> |
+B<--minor> |
+B<--alpha> |
+B<--next-beta> |
+B<--beta> |
+B<--final> |
+B<--branch> |
+B<--no-upload> |
+B<--no-update> |
+B<--verbose> |
+B<--debug> |
+B<--help> |
+B<--manual> |
+]
+
+=head1 DESCRIPTION
+
+B<release.sh> creates an OpenSSL release, given current worktree conditions.
+It will refuses to work unless the current branch is C<master> or a release
+branch (see L</RELEASE BRANCHES AND TAGS> below for a discussion on those).
+
+B<release.sh> tries to be smart and figure out the next release if no hints
+are given through options, and will exit with an error in ambiguous cases.
+
+B<release.sh> always clones the current workspace into a sub-directory
+named C<< openssl-I<MAJOR>.I<MINOR>.I<PATCH>-tmp >>, where C<< I<MAJOR> >>,
+C<< I<MINOR> >> and C<< I<PATCH> >> are taken from the available version
+information in the source.
+
+=head1 OPTIONS
+
+=over 4
+
+=item B<--major>, B<--minor>
+
+Increment the MAJOR or MINOR version number, and reset the lower numbers
+to zero (for B<--major>, the lower numbers are the I<MINOR> and I<PATCH>
+version numbers, for B<--minor>, the lower numbers is the I<PATCH> version
+number), and state of this branch to become "in development". No actual
+release is done.
+
+=item B<--alpha>, B<--beta>
+
+Set the state of this branch to indicate that alpha or beta releases are
+to be done.
+
+B<--alpha> is only acceptable if the I<PATCH> version number is zero and
+the current state is "in development" or that alpha releases are ongoing.
+
+B<--beta> is only acceptable if the I<PATCH> version number is zero and
+that alpha or beta releases are ongoing.
+
+=item B<--next-beta>
+
+Use together with B<--alpha> to switch to beta releases after the current
+release is done.
```

Comment:
What if we later decide to do another alpha?

Reply:
Tough...  I actually think we need to be strict with ourselves.  But either way, it's also still perfectly possible to simply edit `VERSION` and commit it, with good justification.